data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostAttesterSlashing.java

```diff
@@ -19,33 +19,53 @@
 
-import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.http.Context;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.NodeDataProvider;
-import tech.pegasys.teku.api.schema.AttesterSlashing;
-import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
-import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.spec.datastructures.operations.IndexedAttestation;
 import tech.pegasys.teku.statetransition.validation.InternalValidationResult;
 import tech.pegasys.teku.statetransition.validation.ValidationResultCode;
 
-public class PostAttesterSlashing extends AbstractHandler {
+public class PostAttesterSlashing extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/beacon/pool/attester_slashings";
   private final NodeDataProvider nodeDataProvider;
 
-  public PostAttesterSlashing(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    this(dataProvider.getNodeDataProvider(), jsonProvider);
+  public PostAttesterSlashing(final DataProvider dataProvider, final Spec spec) {
+    this(dataProvider.getNodeDataProvider(), spec);
   }
 
-  public PostAttesterSlashing(final NodeDataProvider provider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
+  public PostAttesterSlashing(final NodeDataProvider provider, final Spec spec) {
+    super(
+        EndpointMetadata.post(ROUTE)
+            .operationId("postAttesterSlashing")
+            .summary("Submit attester slashing object")
+            .description(
+                "Submit signed attestations to the beacon node to be validated and submitted if valid.\n\n"
+                    + "This endpoint does not protected against slashing.")
+            .tags(TAG_BEACON)
+            .requestBodyType(getRequestType(spec.getGenesisSpecConfig()))
+            .response(SC_OK, "Success")
+            .response(
+                SC_BAD_REQUEST,
+                "Invalid attester slashing, it will never pass validation so it's rejected",
```

Comment:
delete `it will never pass` through to end, the message will likely contain more context.